core/src/main/scala/sttp/client/ResponseAs.scala

```diff
@@ -134,16 +136,25 @@ object ResponseAs {
     * Converts a deserialization function, which returns errors of type `E`, into a function where errors are thrown
     * as exceptions, and results are returned unwrapped.
     */
-  def deserializeOrThrow[E, T](doDeserialize: String => Either[E, T]): String => T =
+  def deserializeOrThrow[E: ErrorMessage, T](doDeserialize: String => Either[E, T]): String => T =
     s =>
       doDeserialize(s) match {
         case Left(e)  => throw DeserializationError(s, e)
         case Right(b) => b
       }
 }
 
-sealed abstract class ResponseError[+T] extends Exception {
-  def body: String
+sealed abstract class ResponseError[+T](body: String) extends Exception(body)
+case class HttpError(body: String) extends ResponseError[Nothing](body)
```

Comment:
that should probably mention the status code, if we want have an informative message